kythe/cxx/extractor/bazel_artifact_selector.cc

```diff
@@ -214,19 +238,22 @@ void AspectArtifactSelector::ReadFilesInto(
     return;
   }
 
-  if (auto iter = state_.filesets.find(id); iter != state_.filesets.end()) {
+  if (auto node = state_.filesets.extract(id); !node.empty()) {
     state_.disposed.insert(std::string(id));
-    auto node = state_.filesets.extract(iter);
-    const build_event_stream::NamedSetOfFiles& filesets = node.mapped();
-
-    for (const auto& file : filesets.files()) {
-      files.push_back({
-          .local_path = AsLocalPath(file),
-          .uri = AsUri(file),
-      });
+    const FileSet& fileset = node.mapped();
+    files.reserve(files.size() + fileset.files.size());
+    for (const auto* file : fileset.files) {
+      auto iter = state_.files.find(*file);
+      CHECK(iter != state_.files.end()) << "Attempt to remove a missing file!";
```

Comment:
So if we get bad data we crash? i.e. this is an unrecoverable state?

Reply:
This is unrecoverable. The `files` set owns the data. If we fail to lookup the record from a pointer to that record, we're well in UB territory and thoroughly screwed.